frontend/public/components/markdown-view.tsx

```diff
@@ -9,16 +9,23 @@ import './_markdown-view.scss';
 
 const tableTags = ['table', 'thead', 'tbody', 'tr', 'th', 'td'];
 
-const markdownConvert = (markdown, extensions?: string[]) => {
-  const unsafeHtml = new Converter({
+type ShowdownExtension = {
+  type: string;
+  regex?: RegExp;
+  replace?: (...args: any[]) => string;
+};
```

Comment:
Can we add a `render` prop to each extension such that we have a single means for contributing extensions rather than separating the rendering from the showdown extension?
```suggestion
type ShowdownExtension = {
  type: string;
  regex?: RegExp;
  replace?: (...args: any[]) => string;
  render?: (contentDocument: HTMLDocument, rootSelector: string) => React.ReactNode;
};
```

Reply:
I'm not sure if that's a good idea. We will be rendering the component multiple times with each extension and these extensions are directly fed into the showdown converter.

Reply:
I want to combine `renderExtension` and `extensions` into a single prop on the markdown component.
It's odd that the showdown extension and the rendering of the extension are disconnected.

What if we did this:
```
type MarkdownExtension = {
  showdownExtension: {
    type: string;
    regex?: RegExp;
    replace?: (...args: any[]) => string;
  },
  render?: (contentDocument: HTMLDocument, rootSelector: string) => React.ReactNode;
};
```

Can you explain what you mean by:
> We will be rendering the component multiple times with each extension

Instead of accepting a `renderExtensions` prop we can now loop over the extensions and execute the `render` prop.

Reply:
discussed with sahil and we will look into this in the future